ppx-lib/gen_js_api_ppx.ml

```diff
@@ -875,38 +873,60 @@ let split sep s =
   in
   aux 0 0
 
-let ojs_global = ojs_var "global"
+let jsoo_global = jsoo_pure_js_expr "joo_global_object"
 
-let ojs_get o s =
-  if is_ascii s then
-    ojs "get_prop_ascii" [o; str s]
-  else
-    ojs "get_prop" [o; ojs "string_to_js" [str s]]
+let magic x =
+  app (var ("Obj.magic")) (nolabel ([x])) false
 
-let ojs_set o s v =
-  if is_ascii s then
-    ojs "set_prop_ascii" [o; str s; v]
-  else
-    ojs "set_prop" [o; ojs "string_to_js" [str s]; v]
+let is_ascii s =
+  let exception Break in
+  try
+    String.iter (fun c -> if Char.code c > 127 then raise Break) s;
+    true
+  with Break -> false
+
+let get_prop o s =
+  jsoo "get" [o; if is_ascii s then magic (str s) else ojs "string_to_js" [str s]]
```

Comment:
Is the generated code different when removing the `is_ascii` special case ? I would expect the jsoo compiler to do the job already.

Reply:
It's a bit surprising that they are not compiled exactly in the same way actually. 
(also note that is you replace "console" by "con sole" then they are all compiled as `global[cst_console]` which is not surprising). 
It seems that the first strategy as better potential for "minimization". But maybe it is slower to interpret ? I don't know.